src/Platform/Microsoft.Testing.Platform/Hosts/TestHostBuilder.cs

```diff
@@ -724,6 +734,17 @@ private async Task<ITestFramework> BuildTestFrameworkAsync(TestFrameworkBuilderD
             dataConsumersBuilder.Add(pushOnlyProtocolDataConsumer);
         }
 
+        var abortForMaxFailedTestsExtension = new AbortForMaxFailedTestsExtension(
+            serviceProvider.GetCommandLineOptions(),
+            serviceProvider.GetTestFrameworkCapabilities().GetCapability<IGracefulStopTestExecutionCapability>(),
+            serviceProvider.GetRequiredService<IStopPoliciesService>(),
+            serviceProvider.GetTestApplicationCancellationTokenSource());
+
+        if (await abortForMaxFailedTestsExtension.IsEnabledAsync())
+        {
+            dataConsumersBuilder.Add(abortForMaxFailedTestsExtension);
+        }
```

Comment:
@Evangelink Do we want to register it in ServiceProvider though?

Reply:
I see, it's probably not changing much as it's internal but it might help some other extension points we would build. I am fine if we don't.

@MarcoRossignoli any opinion here?

Reply:
I'm fine with the current code, I prefer to be specific and not have it as service as "internal" feature.